### DISCUSSION TITLE
Including LaTeX macros from a file within a custom format

### DISCUSSION THREAD

**Discussion by allefeld** [quarto-dev/quarto-cli#9020] [Q&A] [answered]
### Description

I am trying to define a custom format `document-html` which automatically includes a list of LaTeX macros from a file `math-definitions.tex`. These macros are simple substitutions (e.g. `\newcommand{\rk}{\operatorname{rank}}`) and should be processed by Pandoc, not by MathJax or another math rendering solution.

In another discussion (https://github.com/quarto-dev/quarto-cli/discussions/2845) the result was that the best way to include macros for Pandoc processing was to use an include shortcode,

```markdown
{{< include math-definitions.tex >}}
```

If I include the file in my custom format, the correct shortcode would be

```markdown
{{< include _extensions/document/math-definitions.tex >}}
```

However, I have not been able to figure out how to configure my custom format to include this snippet of text at the beginning of the Markdown source before it is being processed by Quarto (so that the shortcode actually works)?

***

As an alternative, previously in pure Pandoc I defined a format using a defaults file including the line `input-file: math-definitions.tex`. I tried to do the same in the `document/_extension.yml` file for my custom format:

```yaml
contributes:
  formats:
    html:
      input-file: math-definitions.tex
```

When I process a Quarto document using this custom format, Quarto says

```
pandoc 
  to: html
…  

metadata
…
  input-file: _extensions/document/math-definitions.tex
…
```

Which suggests that the option from my `document/_extension.yml` was modified to include the path to the `_extensions` subfolder and passed to Pandoc. However, If I use one of the defined macros in an equation in the document, it doesn't work.

What am I doing wrong? Is there a way to achieve what I want?

**Comment by allefeld** [comment, marked as answer]
Figured it out.

Apparently, `input-file:` is not passed on to Pandoc by Quarto, but `defaults:` is. I can therefore in my custom format definition write

```yaml
contributes:
  formats:
    html:
      defaults: math-definitions.yaml
```

and include in the format a file `math-definitions.yaml` which contains

```yaml
input-file: _extensions/document/math-definitions.tex
```

**Reply by allefeld** [reply]
I haven't checked whether that works with `qmd` files in subdirectories in a project though.

**Reply by allefeld** [reply]
I'm guessing

```yaml
input-file: ${.}/math-definitions.tex
```

should work in projects.

**Reply by dhodge180** [reply]
Did you work out a way to include comments in your math-definitions.tex file?

I find that comments provided using the % comment symbol get displayed in the text at the top (I'm compiling a website).

**Reply by allefeld** [reply]
No, I don't have any comments in my `math-definitions.tex` file. The file is treated as Pandoc input, not as LaTeX input, therefore I guess you could use HTML comments supported by Pandoc:  `<!-- -->`.
